scripts/install.sh

```diff
@@ -0,0 +1,183 @@
+    echo "Error: Helm chart directory not found at $HELM_CHART_DIR"
+    exit 1
+fi
+
+if [ ! -f "$VALUES_FILE" ]; then
+    echo "Error: values.yaml file not found at $VALUES_FILE"
+    exit 1
+fi
+
+# Default CLI detection
+detect_cli() {
+    if command_exists oc; then
+        CLI="oc"
+    elif command_exists kubectl; then
+        CLI="kubectl"
+    else
+        echo "Error: Neither 'oc' nor 'kubectl' is installed. Please install one of them to proceed."
+        exit 1
+    fi
+}
+
+# Parse command-line arguments
+CLI=""
+ROUTER_BASE=""
+RELEASE_NAME=""
+GENERATE_NAME=false
+NAMESPACE=""
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --cli)
+            CLI="$2"
+            if [[ "$CLI" != "oc" && "$CLI" != "kubectl" ]]; then
+                echo "Error: Invalid value for --cli. Use 'oc' or 'kubectl'."
+                exit 1
+            fi
+            shift
+            ;;
+        --router-base)
+            ROUTER_BASE="$2"
+            shift
+            ;;
+        --release-name)
+            RELEASE_NAME="$2"
+            shift
+            ;;
+        --generate-name)
+            GENERATE_NAME=true
+            ;;
+        --namespace)
+            NAMESPACE="$2"
+            shift
+            ;;
+        --help)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "Unknown parameter passed: $1"
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+# If CLI is not specified, detect automatically
+if [[ -z "$CLI" ]]; then
+    detect_cli
+fi
+
+# Check if the user is logged into a cluster
+if ! $CLI whoami >/dev/null 2>&1; then
+    echo "Error: You are not logged into a cluster. Please log in using '$CLI login' or '$CLI config set-context'."
+    exit 1
+fi
+
+# Function to detect cluster router base
+detect_cluster_router_base() {
+    if [ "$CLI" == "oc" ]; then
+        ROUTER_BASE=$($CLI get route -n openshift-console -o=jsonpath='{.items[0].spec.host}')
+    else
+        ROUTER_BASE=$($CLI get ingress -n default -o=jsonpath='{.items[0].spec.rules[0].host}')
+    fi
+}
+
+# Detect cluster router base if not provided
+if [[ -z "$ROUTER_BASE" ]]; then
+    detect_cluster_router_base
+fi
+
+# If detection fails, prompt user for input
+if [[ -z "$ROUTER_BASE" ]]; then
+    echo "Error: Cluster router base could not be detected. Please provide it using the --router-base flag."
+    exit 1
+fi
+
+# Detect namespace if not provided
+if [[ -z "$NAMESPACE" ]]; then
+    NAMESPACE=$($CLI config view --minify --output 'jsonpath={..namespace}')
+    if [[ -z "$NAMESPACE" ]]; then
+        echo "Error: Namespace could not be detected. Please provide it using the --namespace flag."
+        exit 1
+    fi
+fi
+
+# Update Helm chart with the detected or provided router base
+echo "Using router base: $ROUTER_BASE"
+sed -i "s|routerBase:.*|routerBase: $ROUTER_BASE|" "$VALUES_FILE"
+
+# Construct Helm install command
+HELM_CMD="helm install"
```

Comment:
```suggestion
HELM_CMD="helm upgrade --install"
```

Should we attempt to upgrade an existing release, with the option to install if the release does not exist yet?